java/vector/src/main/java/org/apache/arrow/vector/ipc/message/MessageSerializer.java

```diff
@@ -183,6 +183,9 @@ public static ByteBuffer serializeMetadata(Schema schema) {
    * Returns the serialized flatbuffer bytes of the schema wrapped in a message table.
    */
   public static ByteBuffer serializeMetadata(Schema schema, IpcOption writeOption) {
+    if (schema == null) {
+      return ByteBuffer.allocate(0);
+    }
```

Comment:
I think we should adjust callsites instead of implicitly doing something that isn't valid in all contexts (ditto for the two instances below, too)